src/QMCWaveFunctions/RotatedSPOs.h

```diff
@@ -165,15 +165,10 @@ class RotatedSPOs : public SPOSet
 
   void checkInVariables(opt_variables_type& active) override
   {
-    //reset parameters to zero after coefficient matrix has been updated
-    for (int k = 0; k < myVars.size(); ++k)
-      myVars[k] = 0.0;
-
     if (Optimizable)
     {
       if (myVars.size())
         active.insertFrom(myVars);
-      Phi->storeParamsBeforeRotation();
```

Comment:
Correct.
> At each optimizer step, update the rotation parameters and apply the updated (total) rotation matrix to the original coefficients. The rotation parameters always reflect the total rotation.

I'm in favor of this way. The optimization step still generates a rotation matrix delta but RotatedSPO needs to combine it with the accumulated rotation and apply the combined rotation on to the original coefficients by applyRotation.

> At each optimizer step, save a copy of the previous coefficients and set the rotation parameters to zero. The rotation parameters are a delta update at each step. 

This is the way we have in the LCAO code and I would like to move away. applyRotation apply the rotation matrix delta onto the current coefficients.

For the final optimized coefficients, there should be no difference between these ways. However, keeping the "total rotation" instead of final orbitals has the advantages
1. smaller storage. We never want to destroy the orbitals we started with, they are on disk anyway. Then we only need to store the rotation matrix. In a spline case, rotation matrix is just 1k x 1k but storing the final spline coefficients can be GBs.
2. We can do more analysis directly with "total rotation".

So to move from 2 to 1. We need to adjust LCAOrbitalSet and SplineR2R internals.
This also requires updating I/O to make sure the global rotation gets properly handled on disks in the meanwhile, the optimizer only touches the delta rotation matrix.

Reply:
As far as I know, nothing needs to get updated (except the changes in this PR) for the optimizer to always work with the total rotation.
* LCAOrbitalSet::applyRotation applies whatever rotation you give it.  The changes in this PR are to ensure that it operates on the original coefficient matrix
* RotatedSPOs::apply_rotation (which calls LCAOrbitaSet::applyRotation) builds a fresh rotation matrix from the current values of the parameters that are passed in.
* RotatedSPOs::resetParameters takes the new values of the parameters passed to it and passes them to RotatedSPOs::apply_rotation

I don't see where a delta rotation enters in anywhere (that is, after this PR removes setting the parameters to zero in checkInVariables).   Treating the rotation parameters as a total rotation means they behave like any other parameters, and I don't think we need to do anything special to handle them.

Reply:
If parameters are not set to zero, should the RoratedSPO rotation parameter derivatives to be calculated with LCAO using the original coefficients instead of current coefficients?
Probably this is even better. RotatedSPO carries one original SPOSet for derivatives and another SPOSet with rotation for sampling. Then we don't need to fiddle with SPOSet internals.

Reply:
I don't think so.
The evaluateDerivatives function makes no reference to the rotation matrix, so the only way it knows the effects of the rotation parameters on the wavefunction is through the modified coefficient matrix in LCAOrbitalSet.
Also, the tests in #4059 and #4069 use a non-zero value for the rotation and they match the reference values which assume the rotation is applied.

Reply:
If the derivatives are taken with respect to the total rotation parameters, I'm expecting the reference values to be different from current values which are derivatives of the delta rotation parameters.